BrainPortal/lib/cbrain_file_revision.rb

```diff
@@ -99,14 +94,37 @@ def self.[](fullpath)
     self.new(fullpath)
   end
 
-  # Returns the revision info in a fake 'svn' format:
-  #   "$Id: en_cbrain_local_data_provider.rb 12ab34 2010-03-25 17:26:05Z prioux $"
-  # Note that this triggers a self_update() call, if necessary, the first time.
+  # Returns a formatted string representing the last change to the file in git
+  # Format: <commit> <author> <date>
   def to_s
-    return @fake_svn_id_string if ! @fake_svn_id_string.nil? # Cached; no need to fetch git info every time.
+    self.pretty({:file => true, :commit => true, :author => true, :date => true, :time => true})
+  end
+
+  # Returns a formatted string representing the last change to the file in git
+  # Format is determined by pasing attributes that you want as a hash.
+  # Defaults to: <commit> <author> <date>
+  def pretty(rev_fields = {})
```

Comment:
This is the biggest change I suggest: transform pretty() to take a format string in argument:

```ruby
def pretty(format = "%s %a %d")
  etc
```

Supported format characters would be:

* %a = author
* %b = basename
* %c = commit
* %s = short_commit
* %d = date
* %t = time

You can then simply invoke a gsub() on the format string and return the result; gsub() can take a regex as first argument and a hash of subtitution in second args, making this entire new pretty() method a one-liner.